codec-smtp/src/main/java/io/netty/handler/codec/smtp/SmtpResponseDecoder.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2015 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.smtp;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.DecoderException;
+import io.netty.handler.codec.LineBasedFrameDecoder;
+import io.netty.util.CharsetUtil;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Decoder for SMTP responses.
+ */
+public final class SmtpResponseDecoder extends LineBasedFrameDecoder {
+
+    private List<CharSequence> details;
+
+    public SmtpResponseDecoder(int maxLength) {
+        super(maxLength);
```

Comment:
`maxLineLength` just in case a user thinks it's the maximum length of a whole SMTP message?